src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -47,22 +69,134 @@
+
+    if (range != null && range.equals("date"))
+      return getHistoricalPricesForSymbolByDate(symbol, range, date, token);
+    else
+      return getHistoricalPricesForSymbolByRange(symbol, range, token);
+  }
+
+
+  private List<IexHistoricalPrice> getHistoricalPricesForSymbolByDate(final String symbol, final String range, final String date, final String token){
+    if (date == null || date.equals(""))
+      return Collections.emptyList();
+
+    if (date.length() != 8) throw new IllegalArgumentException("Invalid date format.");
+    else {
+      List<IexHistoricalPrice> prices = null;
+      try {
+        LocalDate localDate = LocalDate.of(Integer.parseInt(date.substring(0, 4)),
+            Integer.parseInt(date.substring(4, 6)), Integer.parseInt(date.substring(6, 8)));
+        prices = getHistoricalPricesForSymbolCache(symbol, new ArrayList<>(Arrays.asList(localDate)));
+      }
+      catch (Exception e) {
+        log.info(e.toString());
+      }
+
+      if (prices != null && !prices.isEmpty())
+        return prices;
+      else {
+        List<IexHistoricalPrice> historicalPrice = iexClientExtension.getHistoricalPricesForSymbolByDate(symbol, range, date, token);
+        for (IexHistoricalPrice iexPrice:historicalPrice)
+          historicalPriceRpsy.save(iexPrice);
+        return historicalPrice;
+      }
+    }
+  }
+
+  private List<IexHistoricalPrice> getHistoricalPricesForSymbolByRange(final String symbol, final String range, final String token) {
+    List<IexHistoricalPrice> historicalPrice = new ArrayList<>();
+    LocalDate currentDate = LocalDate.now();
+    LocalDate startingDate = null;
+    List<LocalDate> dates = null;
+    List<IexHistoricalPrice> prices = null;
+
+    if (range == null || range.equals(""))
+      startingDate = checkStartingDate(currentDate, RANGE_DEFAULT);
+    else
+      startingDate = checkStartingDate(currentDate, range);
+
+    if (startingDate == null)
+      throw new IllegalArgumentException("Invalid range over date: Null pointer for date.");
+    log.info("Logging: Today = " + currentDate.toString() + " startingDate = " + startingDate.toString());
+    dates = startingDate.datesUntil(currentDate, Period.ofDays(1)).collect(Collectors.toList());
+    prices = getHistoricalPricesForSymbolCache(symbol, dates);
+    if (!prices.isEmpty())
+      return prices;
     else
-      if(date == null || date.equals(""))
+      return getHistoricalPricesForSymbolIEX(symbol, range, token);
+  }
+
+  private LocalDate checkStartingDate(LocalDate today, String range){
+    if (range.equals(RANGE_YEAR_TO_DATE))
+      return LocalDate.of(today.getYear(), 1, 1);
+    else if (range.equals(RANGE_MAX))
+      return today.minusYears(MAX_YEAR);
+
+    String[] range_part = range.split("(?<=\\d)(?=\\D)");
+    int time_num = 0;
+    String time_span = null;
+    try {
+      time_num= Integer.parseInt(range_part[0]);
+      time_span = range_part[1];
+    }
+    catch (Exception e) {
```

Comment:
This is generally bad form. Should we continue processing if we have an exception here?